rest/importtest/import_test.go

```diff
@@ -3000,3 +3000,105 @@ func TestImportFilterTimeout(t *testing.T) {
 	timeoutErr := rest.WaitWithTimeout(&syncFnFinishedWG, time.Second*15)
 	assert.NoError(t, timeoutErr)
 }
+
+func TestImportRollback(t *testing.T) {
+
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server - needs cbgt and import checkpointing")
+	}
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyImport, base.KeyDCP)
+
+	ctx := base.TestCtx(t)
+	bucket := base.GetPersistentTestBucket(t)
+	defer bucket.Close(ctx)
+
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: bucket.NoCloseClone(),
+		PersistentConfig: false,
+	})
+
+	key := "importRollbackTest"
+	lastSeq := "0"
+	var checkpointPrefix string
+	// do some setup work in an anonymous function so that we can be sure to close the rest tester on unexpected error
+	func() {
+		defer rt.Close()
+
+		// Create a document
+		added, err := rt.GetSingleDataStore().AddRaw(key, 0, []byte(fmt.Sprintf(`{"star": "6"}`)))
+		require.True(t, added)
+		require.NoError(t, err)
+
+		// wait for doc to be imported
+		changes, err := rt.WaitForChanges(1, "/{{.keyspace}}/_changes?since="+lastSeq, "", true)
+		require.NoError(t, err)
+		log.Printf("got changes: %v", changes)
+		var ok bool
+		lastSeq, ok = changes.Last_Seq.(string)
+		require.True(t, ok)
+
+		// Close db while we mess with checkpoints
+		db := rt.GetDatabase()
+		checkpointPrefix = rt.GetDatabase().MetadataKeys.DCPCheckpointPrefix(db.Options.GroupID)
+	}()
+
+	// fetch the checkpoint for the document's vbucket
+	vbNo, err := base.GetVbucketForKey(bucket, key)
+	require.NoError(t, err)
+	log.Printf("got vbno: %v", vbNo)
+
+	// modify the checkpoint to a seq
+
+	metaStore := bucket.GetMetadataStore()
+	checkpointKey := fmt.Sprintf("%s%d", checkpointPrefix, vbNo)
```

Comment:
on a fast system,  can't this run before the goroutine above finishes

Reply:
We don't have anything above in a separate goroutine, just an anonymous function for the defer.